### PR TITLE
bpo-40086: remove test_etree test case

### PR DESCRIPTION
**Pull request by furkanonder**
https://bugs.python.org/issue40086

### CODE REVIEW

**Comment by furkanonder** [comment]
Since cElementTree was deprecated and removed in Python 3 with 36543. So this test is now skipped. So this test had to be removed.

furkan@manjaro ~/cpython$ ./python -m test test_typing -m test_etree -vvv                                                                                                                 == CPython 3.9.0a5+ (heads/master:62d21c9d90, Mar 27 2020, 16:10:32) [GCC 9.3.0]
== Linux-5.5.11-1-MANJARO-x86_64-with-glibc2.31 little-endian
== cwd: /home/furkan/cpython/build/test_python_20035
== CPU count: 8
== encodings: locale=UTF-8, FS=utf-8
0:00:00 load avg: 1.67 Run tests sequentially
0:00:00 load avg: 1.67 [1/1] test_typing
test_etree (test.test_typing.UnionTests) ... skipped 'cElementTree not found'

----------------------------------------------------------------------

Ran 1 test in 0.000s

OK (skipped=1)

== Tests result: SUCCESS ==

1 test OK.

Total duration: 183 ms
Tests result: SUCCESS

**Review by serhiy-storchaka — COMMENTED** [review]
I think that it would be more correct to not remove the test, but making it working with ElementTree (which uses the C implementation by default).

**Comment by furkanonder** [comment]
> I think that it would be more correct to not remove the test, but making it working with ElementTree (which uses the C implementation by default).

Such a note was written in the test function,Only relevant for Python 2.Wouldn't it be more accurate to remove the test since we are not using python2?

**Comment by serhiy-storchaka** [comment]
If it is only relevant for Python 2, why it was added at first place? There is no test_typing in Python 2.

**Comment by ilevkivskyi** [comment]
> I think that it would be more correct to not remove the test, but making it working with ElementTree (which uses the C implementation by default).

+1

I think it makes sense to keep the test. IIRC the idea was to test that C functions don't cause runtime exceptions if used as types.

File: Lib/test/test_typing.py [hunk 1]
```diff
@@ -362,9 +362,9 @@ def test_etree(self):
         # See https://github.com/python/typing/issues/229
         # (Only relevant for Python 2.)
         try:
-            from xml.etree.cElementTree import Element
+            from xml.etree.ElementTree import Element
         except ImportError:
-            raise SkipTest("cElementTree not found")
+            raise SkipTest("ElementTree not found")
```
Comment (serhiy-storchaka) [inline Lib/test/test_typing.py]: I do not think it is possible.

Comment (furkanonder) [reply Lib/test/test_typing.py]: So how can I update? I would be very happy if you could help.

Comment (serhiy-storchaka) [reply Lib/test/test_typing.py]: Remove try/except.

**Review by serhiy-storchaka — APPROVED** [review]

File: Lib/test/test_typing.py (hunk 1 above)
Comment (furkanonder) [reply Lib/test/test_typing.py]: Thanks!

**Review by serhiy-storchaka — COMMENTED** [review]
Please keep the comment.

**Comment by furkanonder** [comment]
> Please keep the comment.

Ok

File: Misc/NEWS.d/next/Tests/2020-03-27-13-29-31.bpo-40086.YNPe8R.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+This test was skipped because cElementTree was deprecated and removed in Python 3 with 36543. So the skipped test was removed.
```
Comment (ilevkivskyi) [inline Misc/NEWS.d/next/Tests/2020-03-27-13-29-31.bpo-40086.YNPe8R.rst]: This comment should be updated.

**Comment by ilevkivskyi** [comment]
Also please see that patchcheck failed because of whitespace issue(s) https://dev.azure.com/Python/cpython/_build/results?buildId=60185&view=logs&j=256d7e09-002a-52d7-8661-29ee3960640e&t=3d7276d3-4e8d-5309-55ad-fb0b172d9925

File: Misc/NEWS.d/next/Tests/2020-03-27-13-29-31.bpo-40086.YNPe8R.rst (hunk 2 above)
Comment (furkanonder) [reply Misc/NEWS.d/next/Tests/2020-03-27-13-29-31.bpo-40086.YNPe8R.rst]: Thanks.It is fixed.

**Review by serhiy-storchaka — COMMENTED** [review]
We usually do not add NEWS entries for minor test changes. They do not affect the end Python user and do not have a useful information for they.

Especially since "This test was skipped" is not informative at all.

**Comment by furkanonder** [comment]
> We usually do not add NEWS entries for minor test changes. They do not affect the end Python user and do not have a useful information for they.
> 
> Especially since "This test was skipped" is not informative at all.

Should i delete my NEWS file?

**Comment by serhiy-storchaka** [comment]
Please delete it.

**Review by ilevkivskyi — APPROVED** [review]

**Pull request merged**